Add migration guide to README file (#631)
Rebilly/rebilly-php: README.md

[ico-license]: https://img.shields.io/badge/license-MIT-brightgreen.svg

[link-api-doc]: https://api-reference.rebilly.com/
[link-github]: https://github.com/Rebilly/rebilly-php

## Migration from SDK v2.x to v3.x

This new major version brings several new helper classes to communicate with Rebilly REST APIs and some incompatibilities that should be tested before switching SDK versions.

- Now, a service class can be explicitly created to make the API calls instead of using only the `Client`. The service is a helper class that encapsulates the API calls for a specific context. It means that `CoreService` is responsible for the API calls related to the [core API](https://api-reference.rebilly.com/), `UsersService` for the [users API](https://user-api-docs.rebilly.com/), and `ReportsService` for the [reports API](https://reports-api-docs.rebilly.com/). `CombinedService` can be used to make calls to any of the contexts.

> - To have a similar code to the previous version, the `Client` object can be straightforwardly used to make calls to any API context. For example, instead of instantiating the service class, it can be just `$client->customers()->create($form)`.

Reviewer: ```suggestion
- To simplify the migration from the previous version, the `Client` object can be used to make calls to any API context directly. For example, instead of instantiating a service class, one can use `$client->customers()->create($form)`.
```

> - The namespace changed from `Rebilly\Entities` to `Rebilly\Sdk\Model`. Now, the `src` is `Rebilly\Sdk\`.

Reviewer: ```suggestion
- The namespace changed from `Rebilly\Entities` to `Rebilly\Sdk\Model`. Now, the `src` is `Rebilly\Sdk`.
```

> - The classes are now generated by an automation process that is triggered every time a new version of the [API definitions](https://github.com/Rebilly/api-definitions) is released. It means more frequency on the SDK updates, keeping it synced with the REST APIs.

Reviewer: ```suggestion
- The classes are now generated by an automated process that is triggered every time a new version of the [API definitions](https://github.com/Rebilly/api-definitions) is released. It means more frequency on the SDK updates, keeping it synced with the REST APIs.
```
